core/shared/src/main/scala/org/http4s/multipart/MultipartDecoder.scala

```diff
@@ -138,9 +138,8 @@ private[http4s] object MultipartDecoder {
               .map[Either[DecodeFailure, Multipart[F]]](parts =>
                 Right(Multipart(parts, Boundary(boundary)))
               )
-              .handleError {
-                case e: InvalidMessageBodyFailure => Left(e)
-                case e => Left(InvalidMessageBodyFailure("Invalid multipart body", Some(e)))
```

Comment:
Sorry for being late to the party. But I'd like to express my opinion without claiming to strictness. I could be wrong, but exceptions being thrown by public API become its contact. Here in `MultipartDecoder`, any exception is transformed to `InvalidMessageBodyFailure`. I guess we can't resolve the initial issue without impacting the public contract, but what if we try to minimise it? What I precisely suggest, in short:
1. Move `org.http4s.server.middleware.EntityLimiter.EntityTooLarge` to the `core` module. 
2. Use it in the `EntityLimiter` Middleware.
3. Propagate it directly in the error handling via `org.http4s.InvalidMessageBodyFailure#cause`.

@armanbilge @LaurenceWarne wdyt?

Reply:
So we would maintain existing behaviour except for special casing the `EntityTooLarge` exception?  If that would be preferable I'm happy to do it.

> Propagate it directly in the error handling via org.http4s.InvalidMessageBodyFailure#cause.

I may not be understanding correctly, but `InvalidMessageBodyFailure.message` doesn't appear to use `cause`, so we wouldn't see the specific error (I guess we could change `details`)?  An alternative could be to use `Pull.raiseError[F](InvalidMessageBodyFailure("Entity too large"))` in the middleware, though I guess this would alter existing behaviour also?

Reply:
> but exceptions being thrown by public API become its contact

@danicheg I see what you're saying, but meh. The constraint is `Concurrent[F]` so users should be prepared for any `Throwable`. If the API was `GenConcurrent[F, InvalidMessageBodyFailure]` then I agree we could not change that explicit contract, but that's not the case. This was just a bug.